src/analysis/special_functions/trigonometric/inverse.lean

```diff
@@ -318,4 +329,25 @@ by { rw [arccos, ← pi_div_four_le_arcsin], split; { intro, linarith } }
 @[continuity]
 lemma continuous_arccos : continuous arccos := continuous_const.sub continuous_arcsin
 
+-- The junk values for `arccos` and `sqrt` make this true even outside `[-1, 1]`.
+lemma tan_arccos (x : ℝ) : tan (arccos x) = sqrt (1 - x ^ 2) / x :=
+by rw [arccos, tan_pi_div_two_sub, tan_arcsin, inv_div]
+
+-- The junk values for `arccos` and `sqrt` make this true even for `1 < x`.
+lemma arccos_eq_arcsin {x : ℝ} (h : 0 ≤ x) :
+  arccos x = arcsin (sqrt (1 - x ^ 2)) :=
+(arcsin_eq_of_sin_eq (sin_arccos _)
+  ⟨(left.neg_nonpos_iff.2 (div_nonneg pi_pos.le (by norm_num))).trans (arccos_nonneg _),
+   arccos_le_pi_div_two.2 h⟩).symm
+
+-- The junk values for `arcsin` and `sqrt` make this true even for `1 < x`.
+lemma arcsin_eq_arccos {x : ℝ} (h : 0 ≤ x) :
+  arcsin x = arccos (sqrt (1 - x ^ 2)) :=
+begin
+  rw eq_comm,
+  convert arccos_cos (arcsin_nonneg.2 h)
+                     ((arcsin_le_pi_div_two _).trans (div_le_self pi_pos.le (by norm_num))),
+  exact (cos_arcsin _).symm
```

Comment:
Optional golfing:
```suggestion
  rw [eq_comm, ← cos_arcsin],
  exact arccos_cos (arcsin_nonneg.2 h)
    ((arcsin_le_pi_div_two _).trans (div_le_self pi_pos.le one_le_two))
```

Otherwise LGTM. Thanks! :tada:

bors d+